### PR TITLE
Fix a bug that st_read_s16 and st_read_f32 reterns the wrong number of frames

### LINKED ISSUE #1201: The wrong number of frames is returned by the st_audiofile function
st_read_s16 and st_read_f32 returns the number of frames multiplied by the number of the WavPack file's channels.

### PR DESCRIPTION
**Pull request by KKQ-KKQ**
fix #1201

### CODE REVIEW

File: external/st_audiofile/src/st_audiofile.c [hunk 1]
```diff
@@ -566,15 +567,16 @@ uint64_t st_read_f32(st_audio_file* af, float* buffer, uint64_t count)
             if (!buf_i32) {
                 return 0;
             }
-            count = channels * WavpackUnpackSamples(af->wv, buf_i32, (uint32_t)count);
-            if (!(af->cache.wv.mode & MODE_FLOAT)) {
+            count = WavpackUnpackSamples(af->wv, buf_i32, (uint32_t)count);
+            {
```
Comment (paulfd) [inline external/st_audiofile/src/st_audiofile.c:571]: I guess from the code above the if condition was always true. If so, we can remove the brackets too.

File: external/st_audiofile/src/st_audiofile.c [hunk 2]
```diff
@@ -566,15 +567,16 @@ uint64_t st_read_f32(st_audio_file* af, float* buffer, uint64_t count)
             if (!buf_i32) {
```
Comment (paulfd) [inline external/st_audiofile/src/st_audiofile.c:567]: For the float case a couple lines above, would we need to consider channel count too?

**Review by paulfd — COMMENTED** [review]
Thanks for the MR! I just had a couple comments.

File: external/st_audiofile/src/st_audiofile.c (hunk 2 above)
Comment (KKQ-KKQ) [reply external/st_audiofile/src/st_audiofile.c:567]: I think the count value was multiplied in order to use for loop.
So we don't have to consider the channel for the float case.

File: external/st_audiofile/src/st_audiofile.c (hunk 1 above)
Comment (KKQ-KKQ) [reply external/st_audiofile/src/st_audiofile.c:571]: Yes, we can.

File: external/st_audiofile/src/st_audiofile.c (hunk 2 above)
Comment (KKQ-KKQ) [reply external/st_audiofile/src/st_audiofile.c:567]: I think we needs it for drwav_f32_to_s16 function.
The function simply converts f32 to s16. so we have to consider the channel count.

**Pull request merged**